common/lib/xmodule/xmodule/discussion_module.py

```diff
@@ -54,9 +69,25 @@ class DiscussionModule(DiscussionFields, XModule):
     js_module_name = "InlineDiscussion"
 
     def get_html(self):
+        course = self.get_course()
+        user = None
+        user_service = self.runtime.service(self, 'user')
+        if user_service:
+            user = user_service._django_user  # pylint: disable=protected-access
+        if user:
+            can_create_comment = has_permission(user, "create_comment", course.id)
+            can_create_subcomment = has_permission(user, "create_sub_comment", course.id)
+            can_create_thread = has_permission(user, "create_thread", course.id)
+        else:
+            can_create_comment = True
+            can_create_subcomment = True
+            can_create_thread = True
```

Comment:
I think False would be more appropriate for all of these values; if there's no user, then new content should not be created.